Userland/DevTools/SQLStudio/MainWidget.cpp

```diff
@@ -0,0 +1,351 @@
+        if (!editor)
+            return;
+        editor->document().redo();
+        update_editor_actions(editor);
+    });
+
+    auto& toolbar_container = add<GUI::ToolbarContainer>();
+    auto& toolbar = toolbar_container.add<GUI::Toolbar>();
+
+    toolbar.add_action(*m_new_action);
+    toolbar.add_action(*m_open_action);
+    toolbar.add_action(*m_save_action);
+    toolbar.add_action(*m_save_as_action);
+    toolbar.add_separator();
+    toolbar.add_action(*m_copy_action);
+    toolbar.add_action(*m_cut_action);
+    toolbar.add_action(*m_paste_action);
+    toolbar.add_separator();
+    toolbar.add_action(*m_undo_action);
+    toolbar.add_action(*m_redo_action);
+
+    m_tab_widget = add<GUI::TabWidget>();
+    m_tab_widget->set_close_button_enabled(true);
+    m_tab_widget->set_reorder_allowed(true);
+
+    m_tab_widget->on_tab_close_click = [&](auto& widget) {
+        auto editor = dynamic_cast<ScriptEditor*>(&widget);
+        if (!editor)
+            return;
+        auto close_attempt = editor->attempt_to_close();
+        if (close_attempt.is_error()) {
+            GUI::MessageBox::show_error(window(), String::formatted("Failed to save before closing\n{}", close_attempt.release_error()));
+            return;
+        }
+        if (close_attempt.release_value()) {
+            m_tab_widget->remove_tab(widget);
+            update_title();
+        }
+    };
+
+    m_tab_widget->on_change = [&](auto&) {
+        update_title();
+        on_editor_change();
+    };
+
+    m_statusbar = add<GUI::Statusbar>(3);
+
+    m_statusbar->segment(1).set_mode(GUI::Statusbar::Segment::Mode::Fixed);
+    m_statusbar->segment(1).set_fixed_width(font().width("000000 characters (00000 words) selected") + font().max_glyph_width());
+
+    m_statusbar->segment(2).set_mode(GUI::Statusbar::Segment::Mode::Fixed);
+    m_statusbar->segment(2).set_fixed_width(font().width("Ln 0000, Col 000") + font().max_glyph_width());
+}
+
+void MainWidget::initialize_menu(GUI::Window* window)
+{
+    auto& file_menu = window->add_menu("&File");
+    file_menu.add_action(*m_new_action);
+    file_menu.add_action(*m_open_action);
+    file_menu.add_action(*m_save_action);
+    file_menu.add_action(*m_save_as_action);
+    file_menu.add_action(*m_save_all_action);
+    file_menu.add_separator();
+    file_menu.add_action(GUI::CommonActions::make_quit_action([](auto&) {
+        GUI::Application::the()->quit();
+    }));
+
+    auto& edit_menu = window->add_menu("&Edit");
+    edit_menu.add_action(*m_copy_action);
+    edit_menu.add_action(*m_cut_action);
+    edit_menu.add_action(*m_paste_action);
+    edit_menu.add_separator();
+    edit_menu.add_action(*m_undo_action);
+    edit_menu.add_action(*m_redo_action);
+
+    auto& help_menu = window->add_menu("&Help");
+    help_menu.add_action(GUI::CommonActions::make_help_action([](auto&) {
+        Desktop::Launcher::open(URL::create_with_file_protocol("/usr/share/man/man1/SQLStudio.md"), "/bin/Help");
```

Comment:
This man page is being introduced in a separate commit, if you want to keep it that way, please also move the code accessing the file to that commit.